### PR TITLE
[mtsource] Automatically adjust Sarama resource usage to fit within l…

### PR DESCRIPTION
**Pull request by lionelvillard**
cherry-pick of #470

### CODE REVIEW

File: pkg/source/client/client.go [hunk 1]
```diff
@@ -119,3 +119,12 @@ func NewProducer(ctx context.Context) (sarama.Client, error) {
 
 	return sarama.NewClient(bs, cfg)
 }
+
+// NewProducer is a helper method for constructing an admin client
+func MakeAdminClient(ctx context.Context, env *KafkaEnvConfig) (sarama.ClusterAdmin, error) {
```
Comment (matzew) [inline pkg/source/client/client.go:124]: how about here ? https://github.com/knative-sandbox/eventing-kafka/blob/2b185a9b59347cc4a16471ed0fb2ba3bdf6e1672/pkg/common/client/admin_client.go#L10

Comment (lionelvillard) [reply pkg/source/client/client.go:124]: yes you had the same comment on the original PR. Not the same function signature.

**Review by matzew — APPROVED** [review]
/lgtm
/approve

**Comment by lionelvillard** [comment]
no timeout for `pull-knative-sandbox-eventing-kafka-upgrade-tests`?

**Pull request merged**